cloudflare/resource_cloudflare_api_token.go

```diff
@@ -0,0 +1,248 @@
+package cloudflare
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func resourceCloudflareApiToken() *schema.Resource {
+	p := schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"resources": {
+				Type:     schema.TypeMap,
+				Required: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"permission_groups": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"effect": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "allow",
+				ValidateFunc: validation.StringInSlice([]string{"allow", "deny"}, false),
+			},
+		},
+	}
+
+	return &schema.Resource{
+		Create: resourceCloudflareApiTokenCreate,
+		Read:   resourceCloudflareApiTokenRead,
+		Update: resourceCloudflareApiTokenUpdate,
+		Delete: resourceCloudflareApiTokenDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"policy": {
+				Type:     schema.TypeList,
+				Optional: true,
```

Comment:
It looks like at least once policy is required so I think the schema should reflect that.